compiler/next/lib/queries/Context.cpp

```diff
@@ -279,10 +307,18 @@ void Context::markUniqueCString(const char* s) {
     buf -= UNIQUED_STRING_METADATA_BYTES; // find start of metadata
     buf += UNIQUED_STRING_METADATA_LEN; // pass the length
     buf[0] = gcMark;
-    assert(buf[1] == 0x02);
+    assert(buf[1] <= 1);        // do-not-collect bit is 0 or 1
```

Comment:
can we have `assert(0 <= buf[1] && buf[1] <= 1); ` to also check it is not negative?